lib/commands/project/project.js

```diff
@@ -383,6 +384,45 @@ function testGenerate(directory, options, cb) {
   var testModule = options.testModule || TESTMODULES[0];
   var assertionFormat = options.assertionFormat || ASSERTIONTYPES[0];
 
+  findProjectFile(directory, null, function(err, projPath) {
+    var projectFile = fs.readFileSync(projPath, 'utf8');
+    var projectJson = JSON.parse(projectFile);
+    var runInstall = false;
+
+    if (!projectJson.devDependencies.hasOwnProperty('z-schema')) {
+      projectJson.devDependencies['z-schema'] = '^3.12.0';
+      runInstall = true;
+    }
+
+    if (!projectJson.devDependencies.hasOwnProperty('chai')) {
+      projectJson.devDependencies.chai = '^3.0.0';
+      runInstall = true;
+    }
+
+    if (!projectJson.devDependencies.hasOwnProperty('mocha')) {
+      projectJson.devDependencies.mocha = '^2.2.5';
+      runInstall = true;
+    }
+
+    if (!projectJson.hasOwnProperty('scripts') || !projectJson.scripts.hasOwnProperty('test')) {
+      projectJson.scripts = {test: './node_modules/mocha/bin/mocha test'};
```

Comment:
In scripts you shouldn't specify a path. It will no work in Windows. npm will use `./node_modules/.bin` as PATH. Here `{"test": "mocha test"}` should be enough. Learn more [here](https://docs.npmjs.com/misc/scripts)